drivers/motor_driver/motor_driver.c

```diff
@@ -29,155 +26,174 @@
-    else if (motor_driver_conf->mode == MOTOR_DRIVER_1_DIR_BRAKE) {
-        if (!gpio_is_valid(dev->gpio_dir0) || \
-            !gpio_is_valid(dev->gpio_dir1_or_brake)) {
-            err = ENODEV;
-            goto motor_set_err;
-        }
-        switch (direction) {
-            case MOTOR_CW:
-            case MOTOR_CCW:
-                /* Direction */
-                gpio_dir0_value = direction;
-                /* No brake */
-                gpio_dir1_or_brake_value = dev->gpio_brake_invert;
-                break;
-            default:
-                pwm_duty_cycle = 0;
-                break;
-        }
-    }
-    else {
-        err = EINVAL;
-        goto motor_set_err;
+    direction = direction ^ motor->gpio_dir_reverse;
+
+    switch (direction) {
+        case MOTOR_CW:
+        case MOTOR_CCW:
+            break;
+        default:
+            pwm_duty_cycle = 0;
+            break;
     }
 
     /* Absolute value of pwm_duty_cycle */
     int32_t pwm_duty_cycle_abs = pwm_duty_cycle;
     pwm_duty_cycle_abs *= (pwm_duty_cycle < 0) ? -1 : 1;
 
-    unsigned irqstate = irq_disable();
-    gpio_write(dev->gpio_dir0, gpio_dir0_value);
-    gpio_write(dev->gpio_dir1_or_brake, gpio_dir1_or_brake_value);
-    pwm_set(motor_driver_conf->pwm_dev, dev->pwm_channel, \
+    /* Critical section */
+    int state = irq_disable();
+
+    /* Set direction */
+    switch (motor_driver->params->mode) {
+        /* Two direction GPIO, handling brake */
+        case MOTOR_DRIVER_2_DIRS:
+            _motor_set_two_dirs(motor, direction);
+        break;
+        /* Single direction GPIO */
+        case MOTOR_DRIVER_1_DIR:
+        case MOTOR_DRIVER_1_DIR_BRAKE:
+            _motor_set_one_dir(motor, direction);
+        break;
+        /* Error */
+        default:
+            LOG_ERROR("Invalid mode to set direction %u\n",
+                      motor_driver->params->mode);
+            err = EINVAL;
+            goto motor_set_err;
+            break;
+    };
+
+    /* Apply PWM duty cycle */
+    pwm_set(motor_driver->params->pwm_dev, motor->pwm_channel, \
             (uint16_t)pwm_duty_cycle_abs);
-    irq_restore(irqstate);
 
-    motor_driver_cb_t cb = motor_driver_conf->cb;
-    if (cb) {
-        cb(motor_driver, motor_id, pwm_duty_cycle);
+    /* Remove brake */
+    switch (motor_driver->params->mode) {
+        /* Two direction GPIO, handling brake */
+        case MOTOR_DRIVER_2_DIRS:
+            _motor_brake_two_dirs(motor, motor_driver->params->brake_inverted);
+        break;
+        case MOTOR_DRIVER_1_DIR:
+        break;
+        case MOTOR_DRIVER_1_DIR_BRAKE:
+            _motor_brake_one_dir_brake(motor, motor_driver->params->brake_inverted);
+        break;
+        /* Error */
+        default:
+            LOG_ERROR("Invalid mode to unbrake %u\n",
+                      motor_driver->params->mode);
+            err = EINVAL;
+            goto motor_set_err;
+            break;
```

Comment:
```suggestion
    /* Two direction GPIO, handling brake */
    case MOTOR_DRIVER_2_DIRS:
        _motor_brake_two_dirs(motor, motor_driver->params->brake_inverted);
        break;
    case MOTOR_DRIVER_1_DIR:
        break;
    case MOTOR_DRIVER_1_DIR_BRAKE:
        _motor_brake_one_dir_brake(motor, motor_driver->params->brake_inverted);
        break;
    /* Error */
    default:
        LOG_ERROR("Invalid mode to unbrake %u\n",
                  motor_driver->params->mode);
        err = EINVAL;
        goto motor_set_err;
        break;
```

As per our Coding Convention: https://github.com/RIOT-OS/RIOT/blob/master/CODING_CONVENTIONS.md#indentation-and-braces